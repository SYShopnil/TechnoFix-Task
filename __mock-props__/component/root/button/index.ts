import { BtnColorSchema, IRedirectButton } from '@src/types/root';

export const mock_data_redirectButton: IRedirectButton[] = [
  {
    btnText: 'Button Link',
    colorSchema: BtnColorSchema.SolidBgGrayTextViolet,
    isOpenNewTab: true,
    redirectLink: 'https://chat.openai.com/',
    dataTestId: 'Redirect Button One',
    isArrow: true,
  },
  {
    btnText: 'Back to Homepage Two',
    colorSchema: BtnColorSchema.SolidBgVioletTextWhite,
    isOpenNewTab: false,
    redirectLink: '',
    dataTestId: 'Redirect Button Two',
    isArrow: false,
  },
];
