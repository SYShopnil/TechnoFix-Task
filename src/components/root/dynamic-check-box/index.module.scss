.fade-container {
  opacity: 0;
  transition: opacity 5s ease-in-out;
}

.fade-in {
  opacity: 1;
}

.fade-out {
  opacity: 0;
  pointer-events: none; /* Prevent interactions when not visible */
}
