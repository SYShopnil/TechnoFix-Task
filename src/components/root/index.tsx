export { IconStore } from './_icon';
export { Button } from './button';
export { DynamicCheckBox } from './dynamic-check-box';
