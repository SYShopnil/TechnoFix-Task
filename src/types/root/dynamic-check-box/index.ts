export interface IDynamicCheckBox {
  title: string;
  checkBoxData: {
    name: string;
    isCheck: boolean;
  }[];
  checkBoxHandler: (index: number) => void;
  dataTestId?: string;
}
