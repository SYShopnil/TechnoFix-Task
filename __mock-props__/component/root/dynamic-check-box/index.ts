import { IDynamicCheckBox } from '@src/types/root';

export const checkBoxItemsDataMock: {
  name: string;
  isCheck: boolean;
}[] = [
  {
    isCheck: true,
    name: 'title',
  },
  {
    isCheck: false,
    name: 'categories',
  },
  {
    isCheck: false,
    name: 'price',
  },
  {
    isCheck: true,
    name: 'date',
  },
  {
    isCheck: true,
    name: 'author',
  },
  {
    isCheck: true,
    name: 'status',
  },
  {
    isCheck: true,
    name: 'action',
  },
];
export const dynamicCheckBoxMock: IDynamicCheckBox = {
  checkBoxData: checkBoxItemsDataMock,
  checkBoxHandler: function (index) {
    console.log(index);
  },
  title: 'Add or removes columns',
  dataTestId: 'checkBoxGetter',
};
