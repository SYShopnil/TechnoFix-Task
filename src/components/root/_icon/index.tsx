import { IIconStore, IconEnums } from '@src/types/root';
import React from 'react';

import { IoIosArrowDown } from 'react-icons/io';
import { RiGraduationCapFill } from 'react-icons/ri';
import { MdQuestionMark } from 'react-icons/md';

export const IconStore = ({ iconName, fill }: IIconStore) => {
  switch (iconName) {
    case IconEnums.DownArrow: {
      return <IoIosArrowDown />;
    }
    case IconEnums.GraduationCap: {
      return <RiGraduationCapFill fill={fill} size={55} />;
    }
    case IconEnums.QuestionSign: {
      return <MdQuestionMark fill={fill} size={45} />;
    }
    default: {
      return <IoIosArrowDown />;
    }
  }
};
