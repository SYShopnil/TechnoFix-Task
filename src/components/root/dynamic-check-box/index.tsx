import { IDynamicCheckBox } from '@src/types/root';
import React from 'react';

export const DynamicCheckBox = ({
  title,
  checkBoxData,
  checkBoxHandler,
  dataTestId,
}: IDynamicCheckBox) => {
  return (
    <div
      className={`bg-[#FFFFFF] shadow-primary p-4 space-y-4 `}
      data-testid={dataTestId || 'checkBoxPicker'}
    >
      <div>
        <h1 className={`font-semibold text-black`}>{title}</h1>
      </div>
      <div className={`space-y-8`}>
        {checkBoxData.map((item, ind) => {
          return (
            <div key={ind}>
              <label htmlFor={item.name}>
                <span className={`relative`}>
                  <input
                    type="checkbox"
                    onChange={() => checkBoxHandler(ind)}
                    checked={item.isCheck}
                    className={`appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto`}
                  />
                </span>
                <span
                  className={`ml-2 font-semibold capitalize text-[1.25rem]`}
                >
                  {item.name}
                </span>
              </label>
            </div>
          );
        })}
      </div>
    </div>
  );
};
