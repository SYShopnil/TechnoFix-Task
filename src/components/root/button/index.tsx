import {
  BtnColorSchema,
  IButton,
  INormalButton,
  IRedirectButton,
} from '@src/types/root';
import Link from 'next/link';
import React, { useState, useEffect } from 'react';
import { IconStore } from '../_icon';
import { IconName } from '@src/types/root/_icon';

//Common layout
const BaseButton = ({ btnText, colorSchema, isArrow }: IButton) => {
  const [fillValue, setFillValue] = useState('');
  const appliedBtnMainWrapperStyle = `flex justify-center items-center `;
  const appliedBtnInnerWrapperStyle = `text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem]`;
  let appliedBtnBgColorStyle = '';

  const setRootFillValue = {
    initial: '',
    onHover: '',
  };
  switch (colorSchema) {
    case BtnColorSchema.SolidBgVioletTextWhite: {
      appliedBtnBgColorStyle = `bg-[#7A67EF] hover:bg-[transparent]  duration-[0.5s] text-[#FFFAFF] hover:text-[#7A67EF]  border-[0.063rem] border-primary_300 hover:border-[#7A67EF]`;
      setRootFillValue['onHover'] = '#727272';
      setRootFillValue['initial'] = '#FFFAFF';
      break;
    }
    case BtnColorSchema.SolidBgWhiteTextGreen: {
      appliedBtnBgColorStyle =
        'bg-[#EBF5EC] hover:bg-[#449657]  duration-[0.5s] text-[#449657] hover:text-[#EBF5EC]   border-[0.063rem] hover:border-[#EBF5EC] border-[#449657]';
      setRootFillValue['onHover'] = '#EBF5EC';
      setRootFillValue['initial'] = '#449657';
      break;
    }
    case BtnColorSchema.SolidBgGrayTextViolet: {
      appliedBtnBgColorStyle =
        'bg-[#EDEDED] hover:bg-[#7A67EE]  duration-[0.5s] text-[#7A67EE] hover:text-[#EDEDED]   border-[0.063rem] hover:border-[#EDEDED] border-[#7A67EE]';
      setRootFillValue['onHover'] = '#EDEDED';
      setRootFillValue['initial'] = '#7A67EE';
      break;
    }
    default: {
      appliedBtnBgColorStyle = `bg-[transparent] hover:bg-primary_500 duration-[0.5s] text-primary_500 hover:text-neutral_500 border-[0.063rem] border-primary_500`;
    }
  }

  // handle the first time arrow initialize
  useEffect(() => {
    setFillValue(setRootFillValue['initial']);
  }, []);

  return (
    // button main wrapper
    <span
      className={`${appliedBtnMainWrapperStyle} `}
      onMouseEnter={() => setFillValue(setRootFillValue.onHover)}
      onMouseLeave={() => setFillValue(setRootFillValue.initial)}
    >
      <span
        className={`${appliedBtnInnerWrapperStyle} w-[100%] h-[100%] ${appliedBtnBgColorStyle}   `}
      >
        <span>{btnText}</span>
        {/* btn text part wrapper */}
        {isArrow && (
          <span
            className={`ml-[1rem] h-[1.5rem] w-[1.5rem]  flex justify-center items-center`}
          >
            <IconStore iconName={IconName.DownArrow} fill={fillValue} />
          </span>
        )}
      </span>
    </span>
  );
};

//Normal Button
export const Button = (props: INormalButton) => {
  return (
    <span
      onClick={(
        e: React.MouseEvent<HTMLElement> | React.FormEvent<HTMLFormElement>
      ) => props.clickHandler?.(e)}
      data-testid={props.dataTestId}
    >
      <BaseButton {...props} />
    </span>
  );
};

//Redirect Button
Button.RedirectButton = (props: IRedirectButton) => {
  return (
    <Link href={props.redirectLink} passHref>
      <a
        target={props.isOpenNewTab ? '_blank' : '_self'}
        data-testid={props.dataTestId}
      >
        <BaseButton {...props} />
      </a>
    </Link>
  );
};
