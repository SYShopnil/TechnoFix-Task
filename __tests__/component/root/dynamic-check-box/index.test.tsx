/**
 *  DONE
 *
 *  Check render successfully of not by taking a snapshot
 *
 */

import { dynamicCheckBoxMock } from '@root/__mock-props__/component/root/dynamic-check-box';
import { DynamicCheckBoxDemo } from '@root/stories/components/root/dynamic-check-box/checkbox.stories';
import { render } from '@testing-library/react';

describe(`When Check Box has rendered`, () => {
  it(`Expect it has been rendered successfully`, () => {
    const { getByTestId } = render(
      <DynamicCheckBoxDemo {...dynamicCheckBoxMock} />
    );
    const checkBoxElement = getByTestId(
      dynamicCheckBoxMock.dataTestId || 'checkBoxPicker'
    );
    expect(checkBoxElement).toBeInTheDocument();
    expect(checkBoxElement).toMatchSnapshot();
  });
});
