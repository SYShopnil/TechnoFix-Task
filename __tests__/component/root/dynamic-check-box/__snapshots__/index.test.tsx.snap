// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Check Box has rendered Expect it has been rendered successfully 1`] = `
<div
  class="bg-[#FFFFFF] shadow-primary p-4 space-y-4 "
  data-testid="checkBoxGetter"
>
  <div>
    <h1
      class="font-semibold text-black"
    >
      Add or removes columns
    </h1>
  </div>
  <div
    class="space-y-8"
  >
    <div>
      <label
        for="Title"
      >
        <span
          class="relative"
        >
          <input
            checked=""
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
          <span
            class="absolute top-[-10%] left-[14%]"
          >
            <svg
              fill="#FFFFFF"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Title
        </span>
      </label>
    </div>
    <div>
      <label
        for="Categories"
      >
        <span
          class="relative"
        >
          <input
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Categories
        </span>
      </label>
    </div>
    <div>
      <label
        for="Price"
      >
        <span
          class="relative"
        >
          <input
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Price
        </span>
      </label>
    </div>
    <div>
      <label
        for="Date"
      >
        <span
          class="relative"
        >
          <input
            checked=""
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
          <span
            class="absolute top-[-10%] left-[14%]"
          >
            <svg
              fill="#FFFFFF"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Date
        </span>
      </label>
    </div>
    <div>
      <label
        for="Author"
      >
        <span
          class="relative"
        >
          <input
            checked=""
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
          <span
            class="absolute top-[-10%] left-[14%]"
          >
            <svg
              fill="#FFFFFF"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Author
        </span>
      </label>
    </div>
    <div>
      <label
        for="Status"
      >
        <span
          class="relative"
        >
          <input
            checked=""
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
          <span
            class="absolute top-[-10%] left-[14%]"
          >
            <svg
              fill="#FFFFFF"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Status
        </span>
      </label>
    </div>
    <div>
      <label
        for="Action"
      >
        <span
          class="relative"
        >
          <input
            checked=""
            class="appearance-none w-5 h-5 border-2 border-gray-300 rounded-md checked:bg-red-500 checked:border-red-600 checked:text-white checked:outline-none checked:relative checked:before:block checked:content-[''] checked:w-5 checked:h-5 checked:border-white checked:border-solid checked:mx-auto"
            type="checkbox"
          />
          <span
            class="absolute top-[-10%] left-[14%]"
          >
            <svg
              fill="#FFFFFF"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ml-2 font-semibold capitalize text-[1.25rem]"
        >
          Action
        </span>
      </label>
    </div>
  </div>
</div>
`;
